web_src/js/features/repo-view-file-tree-sidebar.ts

```diff
@@ -0,0 +1,126 @@
+import {createApp, ref} from 'vue';
+import {toggleElem} from '../utils/dom.ts';
+import {GET, PUT} from '../modules/fetch.ts';
+import ViewFileTree from '../components/ViewFileTree.vue';
+import {initMarkupContent} from '../markup/content.ts';
+import {initTargetRepoBranchTagSelector} from './repo-legacy.ts';
+import {initTargetRepoEllipsisButton} from './repo-commit.ts';
+import {initTargetPdfViewer} from '../render/pdf.ts';
+import {initTargetButtons} from './common-button.ts';
+import {initTargetCopyContent} from './copycontent.ts';
+
+async function toggleSidebar(visibility: boolean, isSigned: boolean) {
+  const sidebarEl = document.querySelector('.repo-view-file-tree-sidebar');
+  const showBtnEl = document.querySelector('.show-tree-sidebar-button');
+  const containerClassList = sidebarEl.parentElement.classList;
+  containerClassList.toggle('repo-grid-tree-sidebar', visibility);
+  containerClassList.toggle('repo-grid-filelist-only', !visibility);
+  toggleElem(sidebarEl, visibility);
+  toggleElem(showBtnEl, !visibility);
+
+  if (!isSigned) return;
+
+  // save to session
+  await PUT('/repo/preferences', {
+    data: {
+      show_file_view_tree_sidebar: visibility,
+    },
+  });
+}
+
+async function loadChildren(path: string, recursive?: boolean) {
+  const fileTree = document.querySelector('#view-file-tree');
+  const apiBaseUrl = fileTree.getAttribute('data-api-base-url');
+  const refTypeNameSubURL = fileTree.getAttribute('data-current-ref-type-name-sub-url');
+  const response = await GET(`${apiBaseUrl}/tree/${refTypeNameSubURL}/${path ?? ''}?recursive=${recursive ?? false}`);
+  const json = await response.json();
+  if (json instanceof Array) {
+    return json.map((i) => ({
+      name: i.name,
+      type: i.type,
+      path: i.path,
+      sub_module_url: i.sub_module_url,
+      children: i.children,
+    }));
+  }
+  return null;
+}
+
+async function loadContent() {
+  // load content by path (content based on home_content.tmpl)
+  const response = await GET(`${window.location.href}?only_content=true`);
+  const contentEl = document.querySelector('.repo-home-filelist');
+  contentEl.innerHTML = await response.text();
+  reloadContentScript(contentEl);
+}
+
+function reloadContentScript(contentEl: Element) {
+  contentEl.querySelector('.show-tree-sidebar-button').addEventListener('click', () => {
+    toggleSidebar(true, document.querySelector('.repo-view-file-tree-sidebar').hasAttribute('data-is-signed'));
+  });
+  initMarkupContent();
+  initTargetButtons(contentEl);
+  // initTargetDropdown(contentEl);
+  initTargetPdfViewer(contentEl);
+  initTargetRepoBranchTagSelector(contentEl);
+  initTargetRepoEllipsisButton(contentEl);
+  initTargetCopyContent(contentEl);
```

Comment:
All these "init" related code could be replaced by either:

* `data-global-init="initMarkupContent"`
* `data-global-click="onXxxButtonClick"`

And use consistent code for first time loaded page and dynamically loaded contents.

Reply:
I will first try to make it run correctly in the new way of global-init.